Backpack/StarRating/Classes/BPKStar.m

```diff
@@ -0,0 +1,160 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "BPKStar.h"
+#import <Backpack/Color.h>
+#import <Backpack/Common.h>
+#import <Backpack/Icon.h>
+#import <Backpack/BPKRTLSupport.h>
+
+@interface BPKStar ()
+
+@property (nonatomic) UIImageView *starView;
+@property (nonatomic) UIImageView *halfStarView;
+
+@end
+
+@implementation BPKStar
+
+- (instancetype)initWithSize:(BPKStarSize)size {
+    BPKAssertMainThread();
+    self = [super initWithFrame:CGRectZero];
+    
+    if (self) {
+        [self setupWithSize:size];
+    }
+    
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    BPKAssertMainThread();
+    self = [super initWithFrame:frame];
+    
+    if (self) {
+        [self setupWithSize:BPKStarSizeSmall];
+    }
+    
+    return self;
+}
+
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
+    BPKAssertMainThread();
+    self = [super initWithCoder:aDecoder];
+    
+    if (self) {
+        [self setupWithSize:BPKStarSizeSmall];
+    }
+    
+    return self;
+}
+
+- (void)setupWithSize:(BPKStarSize)size {
+    
+    _starView = [[UIImageView alloc] init];
```

Comment:
Can we use the property version here please?

Reply:
Done